ECMA2Yaml/ECMAHelper/ECMALoader.docs.cs

```diff
@@ -121,6 +123,54 @@ public Docs LoadDocs(XElement dElement)
             };
         }
 
+        /// <summary>Downgrades markdown headers from 1 - 5. So a `#` becomes `##`, but `######` (ie. h6) remains the same.</summary>
+        /// <param name="remarksText">A string of markdown content</param>
+        public static string DowngradeMarkdownHeaders(string remarksText)
+        {
+            if (string.IsNullOrWhiteSpace(remarksText)) return remarksText ?? string.Empty;
+
+            var lines = remarksText.Replace("\r\n", "\n").Split(new[] { '\n' }, StringSplitOptions.None); //handle both unix and windows line endings
+            
+            bool replaceTriggered = false;
+
+            // walk through the content, first adjusting larger headers and moving in reverse
+            for (int headerSize = 5; headerSize > 0; headerSize--)
+                ReplaceTriggered(lines, headerSize, ref replaceTriggered);
+            
+            return replaceTriggered ? string.Join(Environment.NewLine, lines) : remarksText;
+        }
+
+        /// <summary>Modifies the array if a header of the given size is found</summary>
+        private static void ReplaceTriggered(string[] lines, int headerCount, ref bool replaceTriggered)
+        {
+            string headerPrefix = new string('#', headerCount);
```

Comment:
it's better to use a const string array like {"", "#", "##",...} instead of creating a lot of string objects

Reply:
I've changed this to use a static readonly array